### ISSUE TITLE
some bugs regarding internvl_chat\internvl\train\internvl_chat_finetune.py

### ISSUE THREAD

**Issue by Sprinter1999** [OpenGVLab/InternVL#231]
> 大佬 lora微调完后合并的脚本有吗

@zhangfan-algo 大佬您解决了吗

@czczup 我也想请教一下大佬，lora finetune脚本中，训练会在一定步数的iteration和结束训练后，直接使用huggingface的trainer的save_model()，saved后的model是包含原始模型weight和lora weight的。但不能直接用这个saved model ckpt来做推理，结果会直接乱掉belike: \x\x\x\x。

请问有什么比较好的方法进行处理么

_Originally posted by @Sprinter1999 in https://github.com/OpenGVLab/InternVL/issues/54#issuecomment-2139231134_

**Comment by Sprinter1999** [comment]
Hi, I'm using this bash file (internvl_chat\shell\hermes2_yi34b\internvl_chat_v1_2_hermes2_yi34b_448_res_finetune_continue_lora.sh) to finetune InternVL-1.5. After finetuning for 2 epochs, I find the trained model checkpoint dir (we call it `tuned model`) holds to all model weights. The loss seems to be okay which is lower than 0.5.

After debugging, I find the original released pretrained weights occupies 51G, while the fine-tuned model occupies 51.1G, the increased volumn refers to the lora weights (A & B) on my target modules.

We can use the `tuned model` to inference samples, bug we get fully wrong answers (like `\x\x\x\x`). I guess we should extract the lora weights and add them to the original pretrained weights, I wonder if there's some guidance? Thank you a lot for your support since I have an urgent  task to chase.

**Comment by McCarthyjfq** [comment]
Hi, I just want to ask have you figured out a way to merge the weights after using lora finetune through this file: internvl_chat_v1_2_hermes2_yi34b_448_res_finetune_continue_lora.sh? I have been struggling for some days to merge it and the swift library seems not working. Thank you and appreciate your contribution.
